abey79/vpype
# Feature: new fast, GPU-based viewer

#### Description

The goal is to implement a fast, reusable, GPU-based viewer with outline and preview modes (controllable line width/opacity). This viewer will be used for the `show` command initially, and reused for future projects (vsketch, etc.)

The old, matplotlib-based `show` command is renamed `showmpl` and will be deprecated.

#### For beta tester

Installation: `pip install -U git+https://github.com/abey79/vpype@feature-viewer#egg=vpype`

To test:
 - Navigation:
   - trackpad: horizontal/vertical swiping should scroll the view, pinch in/out gesture should zoom in/out
   - mouse: click-and-drag to scroll, mouse-wheel to zoom in/out
- Display with various files, using all modes (Outline, Outline Colourful, Preview, etc.)

#### TODO

- [x] mouse wheel zoom support
- [x] swipe-scroll support
- [x] ~~`read_multilayer_svg` appears to be very slow, to be verified~~ (unclear, dealt with by #113)
- [x] adaptive decimals for mouse coordinates
- [x] rename linewidth to pen_width
- [x] ~~rulers~~ (later: #166)
- [x] ~~zoom to area tool~~ (later: #159)
- [x] ~~measure tool~~ (later: #158)
- [x] mypy support (add to CI/mypy.ini)
- [x] code documentation
- [x] dark mode icons
- [x] toolbar menu button: menu marker glitch when clicked
- [x] add same params as old `show` command
- [x] checkmark missing on Preview menu item
- [x] delete `__main__.py` files
- [x] fix windows installer
- [x] bug with mouse coord: click and drag, mouse goes outside, coord won't reappear

#### Checklist

- [x] feature/fix implemented
- [x] code formatting ok (`black` and `isort`)
- [x] `mypy vpype vpype_cli tests` returns no error
- [x] tests added/updated and `pytest` succeeds
    - [x] command docstring and option/argument `help`
    - [x] README.md updated (Feature Overview)
    - [x] RTD doc updated and building with no error (`make clean && make html` in `docs/`)
